modules/rte_mis_student_tracking/rte_mis_student_tracking.module

```diff
@@ -213,3 +214,125 @@ function rte_mis_student_tracking_get_total_students() {
   }
   return $total_count;
 }
+
+/**
+ * Implements hook_workflow().
+ */
+function rte_mis_student_tracking_workflow($op, WorkflowTransitionInterface $transition, UserInterface $user) {
+  if ($op == 'transition pre' && $transition->wid->getString() == 'student_admission_workflow' && $transition->getTargetEntityTypeId() == 'mini_node' && $transition->getToSid() == 'student_admission_workflow_admitted') {
+    // If the state is modified to admitted,
+    // create a mini node if any existing mini node,
+    // with same details does not exist of student performance.
+    $allocation_mini_node = $transition->getTargetEntity() ?? NULL;
+    // Array to store the values.
+    $details = [];
+    $details['school_entity_id'] = $allocation_mini_node->get('field_school')->getString() ?? NULL;
+    $details['student_entity_id'] = $allocation_mini_node->get('field_student')->getString() ?? NULL;
+    $details['academic_session'] = $details['entry_year'] = $allocation_mini_node->get('field_academic_year_allocation')->getString() ?? NULL;
+    $details['medium'] = $allocation_mini_node->get('field_medium')->getString() ?? NULL;
+    $details['entry_class'] = $details['current_class'] = $allocation_mini_node->get('field_entry_class_for_allocation')->getString() ?? NULL;
+
+    // Load the student detail from the student entity.
+    $student_details_entity = $allocation_mini_node->get('field_student')->referencedEntities() ?? NULL;
+    $student_details_mini_node = $student_details_entity ? reset($student_details_entity) : NULL;
+
+    if ($student_details_mini_node) {
+      $details['student_name'] = $student_details_mini_node->get('field_student_name')->getString() ?? NULL;
+      $details['caste'] = $student_details_mini_node->get('field_caste')->getString() ?? NULL;
+      $details['date_of_birth'] = $student_details_mini_node->get('field_date_of_birth')->getString() ?? NULL;
+      $details['gender'] = $student_details_mini_node->get('field_gender')->getString() ?? NULL;
+      $details['mobile_number'] = $student_details_mini_node->get('field_mobile_number')->value ?? NULL;
+      $details['religion'] = $student_details_mini_node->get('field_religion')->getString() ?? NULL;
+      $details['residential_address'] = $student_details_mini_node->get('field_residential_address')->getString() ?? NULL;
+
+      $student_gaurdian_name = $student_details_mini_node->get('field_guardian_name')->getString() ?? NULL;
+      $student_father_name = $student_details_mini_node->get('field_father_name')->getString() ?? NULL;
+      $student_mother_name = $student_details_mini_node->get('field_mother_name')->getString() ?? NULL;
+
+      $details['parent_name'] = $student_gaurdian_name ?? $student_father_name ?? $student_mother_name ?? NULL;
```

Comment:
@RohanSingh-inno Can't we do this inside a loop?

Reply:
yes we can do it.